src/unit_tests/analysisd/test_decoder_syscollector.c

```diff
@@ -1493,117 +3030,190 @@ void test_syscollector_dbsync_packages_valid_msg_inserted(void **state)
+    }
+}
+
+static const char *valid_empty_address_array_free[] = { "agent 001 netinfo save NULL|NULL|"
+                "NULL|NULL|NULL|NULL|NULL|NULL|NULL|NULL|NULL|NULL|NULL|NULL|NULL|NULL",
+            "agent 001 netproto save NULL|NULL|0|NULL|NULL|NULL",
+            "agent 001 netaddr save NULL|NULL|0|0.0.0.0|255.255.255.255|0.0.0.1",
+            "agent 001 netproto save NULL|NULL|1|NULL|"
+                "NULL|NULL",
+            "agent 001 netaddr save NULL|NULL|1|0000:0000:0000:0:0000:0000:0000:0000|"
+                "0000::|0000:0000:0000:0:0000:0000:0000:0001" };
+
+void test_syscollector_netinfo_net_array_data_free (void **state, const int edge, const bool parse)
+{
+    Eventinfo *lf = *state;
     const char *result = "ok ";
+    const char *result_not_ok = "not ok ";
     int sock = 1;
+    int i = 0;
+
+    for (; i < edge; i++)
+    {
+        expect_any(__wrap_wdbc_query_ex, *sock);
+        expect_string(__wrap_wdbc_query_ex, query, valid_empty_address_array_free[i]);
+        expect_any(__wrap_wdbc_query_ex, len);
+        will_return(__wrap_wdbc_query_ex, result);
+        will_return(__wrap_wdbc_query_ex, 0);
+    }
 
     expect_any(__wrap_wdbc_query_ex, *sock);
-    expect_string(__wrap_wdbc_query_ex, query, query);
+    expect_string(__wrap_wdbc_query_ex, query, valid_empty_address_array_free[i]);
     expect_any(__wrap_wdbc_query_ex, len);
-    will_return(__wrap_wdbc_query_ex, result);
-    will_return(__wrap_wdbc_query_ex, 0);
+    if (parse)
+    {
+        will_return(__wrap_wdbc_query_ex, result_not_ok);
+        will_return(__wrap_wdbc_query_ex, 0);
+    }
+    else
+    {
+        will_return(__wrap_wdbc_query_ex, result);
+        will_return(__wrap_wdbc_query_ex, -1);
+    }
+    expect_string(__wrap__merror, formatted_msg, "Unable to send netinfo message to Wazuh DB.");
 
     int ret = DecodeSyscollector(lf, &sock);
 
-    assert_int_not_equal(ret, 0);
+    assert_int_not_equal(ret, -1);
 }
-void test_syscollector_dbsync_network_iface_valid_msg_inserted(void **state)
+
+void test_syscollector_netinfo_net_data_free (void **state)
 {
-    Eventinfo *lf = *state;
+    size_t count = sizeof(valid_empty_address_array_free) / sizeof(*valid_empty_address_array_free);
 
-    const char *query = "agent 001 dbsync network_iface INSERTED "
-        "{"
-            "\"scan_time\":\"2021/10/29 14:26:24\","
-            "\"name\":\"59\","
-            "\"adapter\":\"60\","
-            "\"type\":\"61\","
-            "\"state\":\"62\","
-            "\"mtu\":\"63\","
-            "\"mac\":\"64\","
-            "\"tx_packets\":\"65\","
-            "\"rx_packets\":\"66\","
-            "\"tx_bytes\":\"67\","
-            "\"rx_bytes\":\"68\","
-            "\"tx_errors\":\"69\","
-            "\"rx_errors\":\"70\","
-            "\"tx_dropped\":\"71\","
-            "\"rx_dropped\":\"72\","
-            "\"checksum\":\"73\","
-            "\"item_id\":\"74\""
-        "}";
+    for (int i = 0; i < count; i++)
```

Comment:
Question: with this approach, are some values being tested more than once?

Reply:
Yeah, sure, but it's the best approach I've found to comply with code coverage.